File: docs/get-started/python-quickstart/code/part-1/src/summarize/gateway.py
# @@@SNIPSTART quickstart-py-part-1-gateway
from flask import Flask, request, jsonify
from resonate.scheduler import Scheduler
from resonate.storage import LocalPromiseStore
from summarize.app import downloadAndSummarize

app = Flask(__name__)

# Create a Resonate Scheduler   
resonate = Scheduler(durable_promise_storage=LocalPromiseStore())
# Register the downloadAndSummarize function with the Resonate scheduler
resonate.register(downloadAndSummarize)

# Define a route handler for the /summarize endpoint
@app.route('/summarize', methods=['POST'])
def summarize_route_handler():
    try:
        # Extract JSON data from the request
        data = request.get_json()
        if 'url' not in data:
            return jsonify({'error': 'URL not provided'}), 400

        # Extract the URL from the request
        url = data['url']
        
        # Run the summarize function asynchronously
        promise = resonate.run(f"summarize-{url}", downloadAndSummarize, url=url)
        
        # Return the result as JSON
        return jsonify({'summary': promise.result()})
    except Exception as e:
        return jsonify({'error': str(e)}), 500

# Define a main function to start the Flask app
def main():
    app.run(host="127.0.0.1", port=5000)
    print("Serving HTTP on port 5000...")

# Run the main function when the script is invoked
if __name__ == '__main__':
    main()
# @@@SNIPEND

File: docs/get-started/typescript-quickstart/code/part-1/src/gateway.ts
// @@@SNIPSTART quickstart-ts-part-1-gateway
import express, { Request, Response } from "express";
import { Resonate, Context } from "@resonatehq/sdk";
import { downloadAndSummarize } from "./app";

// Initialize a Resonate application.
const resonate = new Resonate();

// Register a function as a Resonate function
resonate.register(
  "downloadAndSummarize", // function name
  downloadAndSummarize, // function pointer
  resonate.options({ timeout: 20000 }) // set a total execution timeout of 20 seconds
);

// Start the Resonate application
resonate.start();

// Initialize an Express application.
const app = express().use(express.json());

// Register a function as an Express endpoint
app.post("/summarize", async (req: Request, res: Response) => {
  const url = req.body?.url;
  try {
    // Call the resonate function
    let summary = await resonate.run(
      "downloadAndSummarize", // function name
      `summarize-${url}`, // promise ID
      url // function argument
    );
    res.send(summary);
  } catch (e) {
    res.status(500).send("An error occurred.");
  }
});

// Start the Express application
app.listen(3000, () => {
  console.log("Listening on port 3000");
});
// @@@SNIPEND
